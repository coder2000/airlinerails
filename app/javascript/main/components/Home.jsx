// @flow

import * as React from 'react';

export default function Home() {
  return <div>Hello from Home component</div>;
}
