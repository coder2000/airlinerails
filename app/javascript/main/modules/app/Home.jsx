// @flow
/* eslint react/jsx-one-expression-per-line: off */

import * as React from 'react';
import {
  Hero, HeroHeader, HeroBody, HeroFooter, Container, Title, Button,
} from 'bloomer';

export default function Home() {
  return (
    <Hero isSize="medium" isColor="dark" isFullHeight>
      <HeroHeader>
        <div>The Airline</div>
      </HeroHeader>
      <HeroBody>
        <Container hasTextAlign="centered">
          <Title>The Airline</Title>
        </Container>
      </HeroBody>
      <HeroFooter style={{ marginBottom: '10vh' }}>
        <Container hasTextAlign="centered">
          <Button isColor="info" isSize="large">
            Get Started
          </Button>
        </Container>
      </HeroFooter>
    </Hero>
  );
}
