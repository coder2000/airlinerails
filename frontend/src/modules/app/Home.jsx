// @flow
/* eslint-disable react/jsx-one-expression-per-line */

import * as React from 'react';
import {
  Hero, HeroHeader, HeroBody, HeroFooter, Container, Title, Button,
} from 'bloomer';
import Navigation from '../navigation';

export default function Home() {
  return (
    <Hero isSize="medium" isFullHeight>
      <HeroHeader>
        <Navigation />
      </HeroHeader>
      <HeroBody>
        <Container hasTextAlign="centered">
          <Title>The Airline</Title>
        </Container>
      </HeroBody>
      <HeroFooter style={{ marginBottom: '10vh' }}>
        <Container hasTextAlign="centered">
          <Button isColor="info" isSize="large" href="/start">
            Get Started
          </Button>
        </Container>
      </HeroFooter>
    </Hero>
  );
}
