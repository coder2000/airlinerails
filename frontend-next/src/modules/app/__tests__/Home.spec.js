import * as React from 'react';
import { shallow } from 'enzyme';
import Home from '../Home';

test('Home renders without error', () => {
  const home = shallow(<Home />);

  expect(home).toBeDefined();
});
