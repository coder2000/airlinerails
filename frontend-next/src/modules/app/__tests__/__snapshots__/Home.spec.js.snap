// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render without error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <withHelpersModifiers(HeroBody)>
          <withHelpersModifiers(Container)
            hasTextAlign="centered"
          >
            <withHelpersModifiers(Title)>
              The Airline
            </withHelpersModifiers(Title)>
          </withHelpersModifiers(Container)>
        </withHelpersModifiers(HeroBody)>,
        <withHelpersModifiers(HeroFooter)
          style={
            Object {
              "marginBottom": "10vh",
            }
          }
        >
          <withHelpersModifiers(Container)
            hasTextAlign="centered"
          >
            <withHelpersModifiers(Button)
              href="/start"
              isColor="info"
              isSize="large"
            >
              New Game
            </withHelpersModifiers(Button)>
          </withHelpersModifiers(Container)>
        </withHelpersModifiers(HeroFooter)>,
      ],
      "className": "is-fullheight-with-navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <withHelpersModifiers(Container)
            hasTextAlign="centered"
          >
            <withHelpersModifiers(Title)>
              The Airline
            </withHelpersModifiers(Title)>
          </withHelpersModifiers(Container)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <withHelpersModifiers(Title)>
              The Airline
            </withHelpersModifiers(Title)>,
            "hasTextAlign": "centered",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "The Airline",
            },
            "ref": null,
            "rendered": "The Airline",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <withHelpersModifiers(Container)
            hasTextAlign="centered"
          >
            <withHelpersModifiers(Button)
              href="/start"
              isColor="info"
              isSize="large"
            >
              New Game
            </withHelpersModifiers(Button)>
          </withHelpersModifiers(Container)>,
          "style": Object {
            "marginBottom": "10vh",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <withHelpersModifiers(Button)
              href="/start"
              isColor="info"
              isSize="large"
            >
              New Game
            </withHelpersModifiers(Button)>,
            "hasTextAlign": "centered",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "New Game",
              "href": "/start",
              "isColor": "info",
              "isSize": "large",
            },
            "ref": null,
            "rendered": "New Game",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <withHelpersModifiers(HeroBody)>
            <withHelpersModifiers(Container)
              hasTextAlign="centered"
            >
              <withHelpersModifiers(Title)>
                The Airline
              </withHelpersModifiers(Title)>
            </withHelpersModifiers(Container)>
          </withHelpersModifiers(HeroBody)>,
          <withHelpersModifiers(HeroFooter)
            style={
              Object {
                "marginBottom": "10vh",
              }
            }
          >
            <withHelpersModifiers(Container)
              hasTextAlign="centered"
            >
              <withHelpersModifiers(Button)
                href="/start"
                isColor="info"
                isSize="large"
              >
                New Game
              </withHelpersModifiers(Button)>
            </withHelpersModifiers(Container)>
          </withHelpersModifiers(HeroFooter)>,
        ],
        "className": "is-fullheight-with-navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <withHelpersModifiers(Container)
              hasTextAlign="centered"
            >
              <withHelpersModifiers(Title)>
                The Airline
              </withHelpersModifiers(Title)>
            </withHelpersModifiers(Container)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <withHelpersModifiers(Title)>
                The Airline
              </withHelpersModifiers(Title)>,
              "hasTextAlign": "centered",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "The Airline",
              },
              "ref": null,
              "rendered": "The Airline",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <withHelpersModifiers(Container)
              hasTextAlign="centered"
            >
              <withHelpersModifiers(Button)
                href="/start"
                isColor="info"
                isSize="large"
              >
                New Game
              </withHelpersModifiers(Button)>
            </withHelpersModifiers(Container)>,
            "style": Object {
              "marginBottom": "10vh",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <withHelpersModifiers(Button)
                href="/start"
                isColor="info"
                isSize="large"
              >
                New Game
              </withHelpersModifiers(Button)>,
              "hasTextAlign": "centered",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "New Game",
                "href": "/start",
                "isColor": "info",
                "isSize": "large",
              },
              "ref": null,
              "rendered": "New Game",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
